cmd/rigs/cmd/publish_layers.go

```diff
@@ -32,17 +35,22 @@ var layersCmd = &cobra.Command{
 		}
 		chunksDir := viper.GetString("chunks-dir")
 
-		if chunksDir == "" && c.String() == "" {
+		if chunksDir == "" && !c.Defined() {
```

Comment:
This was the bug. The empty cid string is `b`. Better to use the `Defined()` func. So this was causing the step of adding the data to IPFS to be skipped.